### PR TITLE
feat: add mark all as read functionality

### PR DESCRIPTION
**Pull request by jainpawan21**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
  feature: This pr adds mark all as read functionality in notification-center
- **What is the current behavior?** (You can also link to an open issue here)
#955 
- **What is the new behavior (if this is a feature change)?**
Mark all as read functionality is working ✅
- **Other information**:

### CODE REVIEW

File: apps/api/src/app/widgets/usecases/mark-all-message-as-seen/mark-all-message-as-seen.usecase.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+@Injectable()
+export class MarkAllMessageAsSeen {
+  constructor(
+    private messageRepository: MessageRepository,
+    private queueService: QueueService,
+    @Inject(ANALYTICS_SERVICE) private analyticsService: AnalyticsService
+  ) {}
+
+  async execute(command: MarkAllMessageAsSeenCommand): Promise<number> {
+    const response = await this.messageRepository.markAllUnseenAsSeen(command.subscriberId);
```
Comment (scopsy) [inline apps/api/src/app/widgets/usecases/mark-all-message-as-seen/mark-all-message-as-seen.usecase.ts:17]: Can we add an `_environomentId` to the query, we work in a multi tenant environment and the subscriberId is not unique here :)

File: apps/api/src/app/widgets/usecases/mark-all-message-as-seen/mark-all-message-as-seen.usecase.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    this.queueService.wsSocketQueue.add({
+      event: 'unseen_count_changed',
+      userId: command.subscriberId,
+      payload: {
+        unseenCount: 0,
+      },
+    });
+
+    this.analyticsService.track('Mark all message as seen - [Notification Center]', command.organizationId, {
+      _subscriber: command.subscriberId,
```
Comment (scopsy) [inline apps/api/src/app/widgets/usecases/mark-all-message-as-seen/mark-all-message-as-seen.usecase.ts:28]: Could you remove the `subscriberId` from here? we don't track user defined properties such as the internal userId 🚀

File: apps/api/src/app/widgets/widgets.controller.ts [hunk 3]
```diff
@@ -110,6 +113,18 @@ export class WidgetsController {
     return await this.markMessageAsSeenUsecase.execute(command);
   }
 
+  @UseGuards(AuthGuard('subscriberJwt'))
+  @Post('/messages/seen')
```
Comment (scopsy) [inline apps/api/src/app/widgets/widgets.controller.ts:117]: Could you please add an `e2e` test for the API on this endpoint?

File: libs/dal/src/repositories/message/message.repository.ts [hunk 4]
```diff
@@ -109,6 +109,13 @@ export class MessageRepository extends BaseRepository<MessageEntity> {
     );
   }
 
+  async markAllUnseenAsSeen(subscriberId: string) {
+    return this.update(
+      { _subscriberId: subscriberId, seen: false },
+      { $set: { seen: true, lastSeenDate: new Date() } }
```
Comment (scopsy) [inline libs/dal/src/repositories/message/message.repository.ts:115]: I think a `$multi` flag is missing here? or is it taken care by the base repostiroy?

File: libs/dal/src/repositories/message/message.repository.ts [hunk 5]
```diff
@@ -109,6 +109,13 @@ export class MessageRepository extends BaseRepository<MessageEntity> {
     );
   }
 
+  async markAllUnseenAsSeen(subscriberId: string) {
+    return this.update(
+      { _subscriberId: subscriberId, seen: false },
```
Comment (scopsy) [inline libs/dal/src/repositories/message/message.repository.ts:114]: Also we need to add _environmentId for this query :)

File: packages/notification-center/src/components/notification-center/components/layout/header/Header.tsx [hunk 6]
```diff
@@ -1,44 +1,28 @@
...
-              color: theme.header?.badgeTextColor,
-              fontWeight: 'bold',
-              fontSize: '12px',
-            }}
-            radius={100}
-          >
-            {unseenCount}
-          </Badge>
-        ) : null}
+        {!tabs && <UnseenBadge unseenCount={unseenCount} />}
```
Comment (scopsy) [inline packages/notification-center/src/components/notification-center/components/layout/header/Header.tsx:21]: @ainouzgali could also lease take a look at this PR? 🚀

**Review by scopsy — COMMENTED** [review]
Thank you @jainpawan21 so much for this one, it's super exciting! Added a couple of comments here. Could you also please provide a small video or images on how this works?

File: apps/api/src/app/widgets/usecases/mark-all-message-as-seen/mark-all-message-as-seen.usecase.ts (hunk 1 above)
Comment (jainpawan21) [reply apps/api/src/app/widgets/usecases/mark-all-message-as-seen/mark-all-message-as-seen.usecase.ts:17]: Sure I will add

File: apps/api/src/app/widgets/usecases/mark-all-message-as-seen/mark-all-message-as-seen.usecase.ts (hunk 2 above)
Comment (jainpawan21) [reply apps/api/src/app/widgets/usecases/mark-all-message-as-seen/mark-all-message-as-seen.usecase.ts:28]: Okay

File: apps/api/src/app/widgets/widgets.controller.ts (hunk 3 above)
Comment (jainpawan21) [reply apps/api/src/app/widgets/widgets.controller.ts:117]: Sure, I forgot to add. I will add

File: libs/dal/src/repositories/message/message.repository.ts (hunk 4 above)
Comment (jainpawan21) [reply libs/dal/src/repositories/message/message.repository.ts:115]: Yes is taken care by base repository

**Comment by scopsy** [comment]
@jainpawan21 i've merged changes from the release branch and we have added a cog functionality there. Could you take a look at the current state of the UI? we might need to modify it a bit. Also when it comes to the feeds support worth checking out: https://www.figma.com/file/9jalQvImfHtbbExZPLW3XO/Concepting?node-id=2244%3A28064 too. @ainouzgali what do you think?

**Comment by jainpawan21** [comment]
> 



> @jainpawan21 i've merged changes from the release branch and we have added a cog functionality there. Could you take a look at the current state of the UI? we might need to modify it a bit. Also when it comes to the feeds support worth checking out: https://www.figma.com/file/9jalQvImfHtbbExZPLW3XO/Concepting?node-id=2244%3A28064 too. @ainouzgali what do you think?

Sure @scopsy, I will look into this

**Comment by jainpawan21** [comment]
> @jainpawan21 i've merged changes from the release branch and we have added a cog functionality there. Could you take a look at the current state of the UI? we might need to modify it a bit. Also when it comes to the feeds support worth checking out: https://www.figma.com/file/9jalQvImfHtbbExZPLW3XO/Concepting?node-id=2244%3A28064 too. @ainouzgali what do you think?

Settings icon is missing in mentioned figma design

**Comment by jainpawan21** [comment]
> Thank you @jainpawan21 so much for this one, it's super exciting! Added a couple of comments here. Could you also please provide a small video or images on how this works?

@scopsy, This is how it is working. Unfortunately, real-time updates are not working for me. [Relevant issue](#1008 )

https://user-images.githubusercontent.com/39362422/184659172-0758ca95-c99a-4fb3-b089-99db4eb28486.mov

**Comment by scopsy** [comment]
@jainpawan21 I'm checking with our designer about the mark all as seen location if that's ok. And otherwise we can merge :)

**Comment by jainpawan21** [comment]
Shall I fix merge conflict?

**Comment by jainpawan21** [comment]
Hi @scopsy, Shall I raise PR again for main branch?

**Comment by scopsy** [comment]
Ohhh I deleted the 0.7.0 branch and Github closed this PR automatically :|| Sorry about that, could you please reopen this against main? @jainpawan21

**Pull request closed without merging**